yb-voyager/cmd/migrationStatus.go

```diff
@@ -0,0 +1,37 @@
+package cmd
+
+import (
+	"fmt"
+)
+
+type MigrationStatusRecord struct {
+	MigrationUUID      string
+	SourceDBType       string
+	ExportType         string
+	FallForwarDBExists bool
+}
+
+const MIGRATION_STATUS_KEY = "migration_status"
+
+func UpdateMigrationStatusRecord(updateFn func(*MigrationStatusRecord)) error {
+	return UpdateJsonObjectInMetaDB(metaDB, MIGRATION_STATUS_KEY, updateFn)
+}
+
+func GetMigrationStatusRecord() (*MigrationStatusRecord, error) {
+	record := new(MigrationStatusRecord)
+	found, err := metaDB.GetJsonObject(nil, MIGRATION_STATUS_KEY, record)
+	if err != nil {
+		return nil, fmt.Errorf("error while getting migration status record from meta db: %w", err)
+	}
+	if !found {
+		return nil, nil
+	}
+	return record, nil
+}
+
+func InitMigrationStatusRecord(migUUID string) error {
+	return UpdateMigrationStatusRecord(func(record *MigrationStatusRecord) {
+		record.MigrationUUID = migUUID
+		record.ExportType = SNAPSHOT_ONLY
+	})
```

Comment:
BUG: The current implementation is not idempotent. `CreateMigrationProjectIfNotExists()` is called from every command invocation. So `InitMigrationStatusRecord()` will be called once for every command.

The current implementation will correctly initialise `migUUID` and `ExportType` the first time it runs. On the next run, it will overwrite `migUUID` with empty UUID and reset `ExportType` back to `SNAPSHOT_ONLY`.

One approach to make this function idempotent is to immediately return if `record.MigrationUUID` is not "nil uuid" (there is a const to represent a nil uuid).